libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -588,7 +656,31 @@ private void EnsureDependenciesInstalled()
                         Selector = new MostSpecificSelector { Selector = new FirstSelector() };
                     }
 
-                    Selector.Initialize(Triggers);
+                    if (needsTracker)
+                    {
+                        if (!dc.GetState().ContainsKey(DialogPath.EventCounter))
```

Comment:
>dc.GetState().ContainsKey [](start = 29, length = 25)

This seems a little brittle given that this will only ever get called once for the lifetime of the bot.

Reply:
I don't think I understand what you propose.